Update sample request links to use sandbox

rmglennon opened the pull request:
These changes mostly update the links to use the (awesome!) sandbox site, compared to the URL to the raw JSON response. This is done to make sure the links work following the requirement to use an API key with all requests.

I made a few other minor edits as I went through, such as typos, markdown formatting (there should be a space between a # and the text in a heading and a line break after a heading), and style to change mapzen-xxxxxx to the preferred variable of `your-mapzen-api-key`.

In some cases, these queries give different results now than what we have documented. I have noted those places and we can review later...this is slightly more than a find and replace exercise, but does not include significant content updates.

dianashk commented on search.md:
````diff
@@ -9,7 +9,7 @@ Making the leap from text to coordinates is an intricate and challenging process
 ```
-   https://search.mapzen.com/v1/search?text=London&api_key=mapzen-xxxxxx
+   https://search.mapzen.com/v1/search?text=London&api_key=your-mapzen-api-key
    \___/   \_______________/\__/\_____/\__________/\___________________/
````
should we extend the last line to come to the end of the api key placeholder text for consistency?

dianashk commented:
@rmglennon, you mentioned some results being different, which ones in particular? I'd like to have a look at them to make sure nothing is broken. :) Thanks.

rmglennon commented:
Thanks for the review, @dianashk! I've fixed that little underscore issue.

I dropped my notes in https://github.com/pelias/pelias-doc/issues/194. You can ignore the old/new href lines, which were just for my copy/paste sanity.
